Error in return type for findAll when service is not enabled but DTO is enabled

<!--
> Please follow the issue template below for bug reports and queries.
> Bug reports or queries opened without any of these info will be **closed** without any explanation.
> For feature requests, label the title with [FEATURE] and explain your use case and ideas clearly below, you can remove sections which are not relevant.
> For bug reports your `.yo-rc.json` file is mandatory else we will close the issue.
> You can run `yo jhipster:info` in your project folder to get most of the required info
-->

##### **Overview of the issue**

Error in return type for findAll when service is not enabled but DTO is enabled
```
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /D:/Projects/test/testNew/src/main/java/com/mycompany/myapp/web/rest/JobResource.java:[110,34] incompatible types: cannot infer type arguments for org.springframework.http.ResponseEntity<>
    reason: inference variable T has incompatible bounds
      equality constraints: java.util.List<com.mycompany.myapp.service.dto.JobDTO>
      lower bounds: java.util.List<com.mycompany.myapp.domain.Job>
[ERROR] /D:/Projects/test/testNew/src/main/java/com/mycompany/myapp/web/rest/EmployeeResource.java:[110,34] incompatible types: cannot infer type arguments for org.springframework.http.ResponseEntity<>
    reason: inference variable T has incompatible bounds
      equality constraints: java.util.List<com.mycompany.myapp.service.dto.EmployeeDTO>
      lower bounds: java.util.List<com.mycompany.myapp.domain.Employee>
[INFO] 2 errors
```

##### **Reproduce the error**

Generate a default app use below JDL 
```
entity Region {
	regionId Long,
	regionName String
}

entity Country {
	countryId Long,
	countryName String
}

// an ignored comment
/** not an ignored comment */
entity Location {
	locationId Long,
	streetAddress String,
	postalCode String,
	city String,
	stateProvince String
}

entity Department {
	departmentId Long,
	departmentName String required
}

/**
 * Task entity.
 * @author The JHipster team.
 */
entity Task {
	taskId Long,
	title String,
	description String
}

/**
 * The Employee entity.
 */
entity Employee {
	employeeId Long,
	/**
	* The firstname attribute.
	*/
	firstName String,
	lastName String,
	email String,
	phoneNumber String,
	hireDate ZonedDateTime,
	salary Long,
	commissionPct Long
}

entity Job {
	jobId Long,
	jobTitle String,
	minSalary Long,
	maxSalary Long
}

entity JobHistory {
	startDate ZonedDateTime,
	endDate ZonedDateTime,
	language Language
}

enum Language {
    FRENCH, ENGLISH, SPANISH
}

relationship OneToOne {
	Country{region} to Region
}

relationship OneToOne {
	Location{country} to Country
}

relationship OneToOne {
	Department{location} to Location
}

relationship ManyToMany {
	Job{task(title)} to Task{job}
}

// defining multiple OneToMany relationships with comments
relationship OneToMany {
	Employee{job} to Job,
	/**
	* A relationship
	*/
	Department{employee} to
	/**
	* Another side of the same relationship
	*/
	Employee
}

relationship ManyToOne {
	Employee{manager} to Employee
}

// defining multiple oneToOne relationships
relationship OneToOne {
	JobHistory{job} to Job,
	JobHistory{department} to Department,
	JobHistory{employee} to Employee
}

// Set pagination options
paginate JobHistory, Employee with infinite-scroll
paginate Job with pagination

dto * with mapstruct except Job

// Set service options to all except few
service all with serviceImpl except Employee, Job
// Set an angular suffix
angularSuffix * with mySuffix

```

##### **Related issues**

Related to #3175 fix

##### **Suggest a Fix**

Need to return a mapped DTO from find all

##### **JHipster Version(s)**

Master

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

default app

--- a/generators/entity/templates/src/main/java/package/common/get_all_template.ejs
+++ b/generators/entity/templates/src/main/java/package/common/get_all_template.ejs
@@ -13,11 +13,9 @@
     public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(Pageable pageable<% if (fieldsContainNoOwnerOneToOne) { %>, @RequestParam(required = false) String filter<% } %>)
         throws URISyntaxException {<%- include('get_all_stream_template', {viaService: viaService}); -%>
         log.debug("REST request to get a page of <%= entityClassPlural %>");<% if (viaService) { %>
-        Page<<%= instanceType %>> page = <%= entityInstance %>Service.findAll(pageable);
+        Page<<%= instanceType %>> page = <%= entityInstance %>Service.findAll(pageable);<% } else { %>
+        Page<<%= entityClass %>> page = <%= entityInstance %>Repository.findAll(pageable);<% } %>
         HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/<%= entityApiUrl %>");
-        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);<% } else { %>
-        Page<<%= entityClass %>> page = <%= entityInstance %>Repository.findAll(pageable);
-        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/<%= entityApiUrl %>");
-        return new ResponseEntity<>(<% if (dto == 'mapstruct') { %><%= entityListToDtoListReference %>(page.getContent())<% } else { %>page.getContent()<% } %>, headers, HttpStatus.OK);<% } %>
+        return new ResponseEntity<>(<% if (!viaService && dto == 'mapstruct') { %><%= entityListToDtoListReference %>(page.getContent())<% } else { %>page.getContent()<% } %>, headers, HttpStatus.OK);
     <% } -%>
 }

--- a/generators/entity/templates/src/main/java/package/common/get_all_template.ejs
+++ b/generators/entity/templates/src/main/java/package/common/get_all_template.ejs
@@ -13,9 +13,11 @@
     public ResponseEntity<List<<%= instanceType %>>> getAll<%= entityClassPlural %>(Pageable pageable<% if (fieldsContainNoOwnerOneToOne) { %>, @RequestParam(required = false) String filter<% } %>)
         throws URISyntaxException {<%- include('get_all_stream_template', {viaService: viaService}); -%>
         log.debug("REST request to get a page of <%= entityClassPlural %>");<% if (viaService) { %>
-        Page<<%= instanceType %>> page = <%= entityInstance %>Service.findAll(pageable); <% } else { %>
-        Page<<%= entityClass %>> page = <%= entityInstance %>Repository.findAll(pageable); <% } %>
+        Page<<%= instanceType %>> page = <%= entityInstance %>Service.findAll(pageable);
         HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/<%= entityApiUrl %>");
-        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
+        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);<% } else { %>
+        Page<<%= entityClass %>> page = <%= entityInstance %>Repository.findAll(pageable);
+        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/<%= entityApiUrl %>");
+        return new ResponseEntity<>(<% if (dto == 'mapstruct') { %><%= entityListToDtoListReference %>(page.getContent())<% } else { %>page.getContent()<% } %>, headers, HttpStatus.OK);<% } %>
     <% } -%>
 }

--- a/generators/entity/templates/src/main/java/package/common/search_template.ejs
+++ b/generators/entity/templates/src/main/java/package/common/search_template.ejs
@@ -11,9 +11,9 @@
     public ResponseEntity<List<<%= instanceType %>>> search<%= entityClassPlural %>(@RequestParam String query, Pageable pageable)
         throws URISyntaxException {
         log.debug("REST request to search for a page of <%= entityClassPlural %> for query {}", query);<% if (viaService) { %>
-        Page<<%= entityClass %>> page = <%= entityInstance %>Service.search(query, pageable);<% } else { %>
+        Page<<%= instanceType %>> page = <%= entityInstance %>Service.search(query, pageable);<% } else { %>
         Page<<%= entityClass %>> page = <%= entityInstance %>SearchRepository.search(queryStringQuery(query), pageable);<% } %>
         HttpHeaders headers = PaginationUtil.generateSearchPaginationHttpHeaders(query, page, "/api/_search/<%= entityApiUrl %>");
-        return new ResponseEntity<>(<% if (dto == 'mapstruct') { %><%= entityListToDtoListReference %>(page.getContent())<% } else { %>page.getContent()<% } %>, headers, HttpStatus.OK);
+        return new ResponseEntity<>(<% if (!viaService && dto == 'mapstruct') { %><%= entityListToDtoListReference %>(page.getContent())<% } else { %>page.getContent()<% } %>, headers, HttpStatus.OK);
     <% } -%>
 }

--- a/generators/entity/templates/src/main/java/package/service/_EntityService.java
+++ b/generators/entity/templates/src/main/java/package/service/_EntityService.java
@@ -66,5 +66,5 @@ public interface <%= entityClass %>Service {
      *  @param pageable the pagination information<% } %>
      *  @return the list of entities
      */
-    <% if (pagination != 'no') { %>Page<<%= entityClass %><% } else { %>List<<%= instanceType %><% } %>> search(String query<% if (pagination != 'no') { %>, Pageable pageable<% } %>);<% } %>
+    <% if (pagination != 'no') { %>Page<<%= instanceType %><% } else { %>List<<%= instanceType %><% } %>> search(String query<% if (pagination != 'no') { %>, Pageable pageable<% } %>);<% } %>
 }

--- a/generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java
+++ b/generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java
@@ -105,12 +105,16 @@ public class <%= serviceClassName %> <% if (service == 'serviceImpl') { %>implem
      *  @return the list of entities
      */<% if (databaseType == 'sql') { %>
     @Transactional(readOnly = true)<% } %>
-    public <% if (pagination != 'no') { %>Page<<%= entityClass %><% } else { %>List<<%= instanceType %><% } %>> search(String query<% if (pagination != 'no') { %>, Pageable pageable<% } %>) {
+    public <% if (pagination != 'no') { %>Page<<%= instanceType %><% } else { %>List<<%= instanceType %><% } %>> search(String query<% if (pagination != 'no') { %>, Pageable pageable<% } %>) {
         <%_ if (pagination == 'no') { _%>
         log.debug("Request to search <%= entityClassPlural %> for query {}", query);<%- include('../../common/search_stream_template', {viaService: viaService}); -%>
         <%_ } else { _%>
         log.debug("Request to search for a page of <%= entityClassPlural %> for query {}", query);
-        return <%= entityInstance %>SearchRepository.search(queryStringQuery(query), pageable);
-        <%_ } _%>
+        Page<<%= entityClass %>> result = <%= entityInstance %>SearchRepository.search(queryStringQuery(query), pageable);
+            <%_ if (dto == 'mapstruct') { _%>
+        return result.map(<%= entityInstance %> -> <%= entityToDto %>(<%= entityInstance%>));
+            <%_ } else { _%>
+        return result;
+        <%_ } } _%>
     }<% } %>
 }

--- a/generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java
+++ b/generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java
@@ -71,7 +71,7 @@ public class <%= serviceClassName %> <% if (service == 'serviceImpl') { %>implem
         <%_ } else { _%>
         Page<<%= entityClass %>> result = <%= entityInstance %>Repository.findAll(pageable);
             <%_ if (dto == 'mapstruct') { _%>
-        return result.map(<%= instanceName %> -> <%= entityToDto %>(<%= instanceName%>));
+        return result.map(<%= entityInstance %> -> <%= entityToDto %>(<%= entityInstance%>));
             <%_ } else { _%>
         return result;
         <%_ } } _%>

--- a/travis/samples/.jhipster/EntityWithDTO.json
+++ b/travis/samples/.jhipster/EntityWithDTO.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123735",
+    "entityTableName": "entity_with_dto",
+    "dto": "mapstruct",
+    "pagination": "no",
+    "service": "no"
+}

--- a/travis/samples/.jhipster/EntityWithPagination.json
+++ b/travis/samples/.jhipster/EntityWithPagination.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123738",
+    "entityTableName": "entity_with_pagination",
+    "dto": "no",
+    "pagination": "pagination",
+    "service": "no"
+}

--- a/travis/samples/.jhipster/EntityWithPaginationAndDTO.json
+++ b/travis/samples/.jhipster/EntityWithPaginationAndDTO.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123743",
+    "entityTableName": "entity_with_pagination_and_dto",
+    "dto": "mapstruct",
+    "pagination": "pagination",
+    "service": "no"
+}

--- a/travis/samples/.jhipster/EntityWithServiceClass.json
+++ b/travis/samples/.jhipster/EntityWithServiceClass.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123736",
+    "entityTableName": "entity_with_service_class",
+    "dto": "no",
+    "pagination": "no",
+    "service": "serviceClass"
+}

--- a/travis/samples/.jhipster/EntityWithServiceClassAndDTO.json
+++ b/travis/samples/.jhipster/EntityWithServiceClassAndDTO.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123741",
+    "entityTableName": "entity_with_service_class_and_dto",
+    "dto": "mapstruct",
+    "pagination": "no",
+    "service": "serviceClass"
+}

--- a/travis/samples/.jhipster/EntityWithServiceClassAndPagination.json
+++ b/travis/samples/.jhipster/EntityWithServiceClassAndPagination.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123739",
+    "entityTableName": "entity_with_service_class_and_pagination",
+    "dto": "no",
+    "pagination": "pagination",
+    "service": "serviceClass"
+}

--- a/travis/samples/.jhipster/EntityWithServiceClassPaginationAndDTO.json
+++ b/travis/samples/.jhipster/EntityWithServiceClassPaginationAndDTO.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123744",
+    "entityTableName": "entity_with_service_class_pagination_and_dto",
+    "dto": "mapstruct",
+    "pagination": "pagination",
+    "service": "serviceClass"
+}

--- a/travis/samples/.jhipster/EntityWithServiceImpl.json
+++ b/travis/samples/.jhipster/EntityWithServiceImpl.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123737",
+    "entityTableName": "entity_with_service_impl",
+    "dto": "no",
+    "pagination": "no",
+    "service": "serviceImpl"
+}

--- a/travis/samples/.jhipster/EntityWithServiceImplAndDTO.json
+++ b/travis/samples/.jhipster/EntityWithServiceImplAndDTO.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123742",
+    "entityTableName": "entity_with_service_impl_and_dto",
+    "dto": "mapstruct",
+    "pagination": "no",
+    "service": "serviceImpl"
+}

--- a/travis/samples/.jhipster/EntityWithServiceImplAndPagination.json
+++ b/travis/samples/.jhipster/EntityWithServiceImplAndPagination.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123740",
+    "entityTableName": "entity_with_service_impl_and_pagination",
+    "dto": "no",
+    "pagination": "pagination",
+    "service": "serviceImpl"
+}

--- a/travis/samples/.jhipster/EntityWithServiceImplPaginationAndDTO.json
+++ b/travis/samples/.jhipster/EntityWithServiceImplPaginationAndDTO.json
@@ -1,0 +1,14 @@
+{
+    "relationships": [],
+    "fields": [
+        {
+            "fieldName": "name",
+            "fieldType": "String"
+        }
+    ],
+    "changelogDate": "20160806123745",
+    "entityTableName": "entity_with_service_impl_pagination_and_dto",
+    "dto": "mapstruct",
+    "pagination": "pagination",
+    "service": "serviceImpl"
+}

--- a/travis/scripts/02-generate-entities.sh
+++ b/travis/scripts/02-generate-entities.sh
@@ -76,6 +76,18 @@ elif [[ ("$JHIPSTER" == "app-mysql") || ("$JHIPSTER" == "app-psql-es-noi18n") ]]
   moveEntity TestOneToOne
   moveEntity TestCustomTableName
 
+  moveEntity EntityWithDTO
+  moveEntity EntityWithPagination
+  moveEntity EntityWithPaginationAndDTO
+  moveEntity EntityWithServiceClass
+  moveEntity EntityWithServiceClassAndDTO
+  moveEntity EntityWithServiceClassAndPagination
+  moveEntity EntityWithServiceClassPaginationAndDTO
+  moveEntity EntityWithServiceImpl
+  moveEntity EntityWithServiceImplAndDTO
+  moveEntity EntityWithServiceImplAndPagination
+  moveEntity EntityWithServiceImplPaginationAndDTO
+
 else
   moveEntity BankAccount
   moveEntity Label
@@ -88,6 +100,18 @@ else
   moveEntity FieldTestInfiniteScrollEntity
   moveEntity FieldTestPagerEntity
   moveEntity FieldTestPaginationEntity
+
+  moveEntity EntityWithDTO
+  moveEntity EntityWithPagination
+  moveEntity EntityWithPaginationAndDTO
+  moveEntity EntityWithServiceClass
+  moveEntity EntityWithServiceClassAndDTO
+  moveEntity EntityWithServiceClassAndPagination
+  moveEntity EntityWithServiceClassPaginationAndDTO
+  moveEntity EntityWithServiceImpl
+  moveEntity EntityWithServiceImplAndDTO
+  moveEntity EntityWithServiceImplAndPagination
+  moveEntity EntityWithServiceImplPaginationAndDTO
 fi
 
 ls -l "$HOME"/app/.jhipster/
@@ -129,6 +153,18 @@ generateEntity TestManyToOne
 generateEntity TestManyToMany
 generateEntity TestOneToOne
 generateEntity TestCustomTableName
+
+generateEntity EntityWithDTO
+generateEntity EntityWithPagination
+generateEntity EntityWithPaginationAndDTO
+generateEntity EntityWithServiceClass
+generateEntity EntityWithServiceClassAndDTO
+generateEntity EntityWithServiceClassAndPagination
+generateEntity EntityWithServiceClassPaginationAndDTO
+generateEntity EntityWithServiceImpl
+generateEntity EntityWithServiceImplAndDTO
+generateEntity EntityWithServiceImplAndPagination
+generateEntity EntityWithServiceImplPaginationAndDTO
 
 #-------------------------------------------------------------------------------
 # Check Javadoc generation

